providers/loading.tsx

```diff
@@ -0,0 +1,19 @@
+import { createContext, Dispatch, SetStateAction, useState } from "react";
+import Loading from "../components/loading";
+
+let setStateAction: Dispatch<SetStateAction<boolean>>;
+
+const LoadingContext = createContext(setStateAction);
```

Comment:
`setStateAction` is being used as a default value, but the default value typed does not match the value that LoadingContext will have, nor is the variable initialized with it - so you're initializing this state with an undefined default state, `createContext(false)` is more than enough